recipes/libsndfile/all/conanfile.py

```diff
@@ -0,0 +1,104 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment, RunEnvironment
+from conans.errors import ConanInvalidConfiguration
+import os
+import shutil
+import glob
+
+
+class LibnameConan(ConanFile):
+    name = "libsndfile"
+    description = "Libsndfile is a library of C routines for reading and writing files containing sampled audio data."
+    topics = ("conan", "libsndfile", "sound")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://www.mega-nerd.com/libsndfile"
+    license = "LGPL-2.1"
+    generators = 'pkg_config'
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_alsa": [True, False],
+        "with_sqlite": [True, False],
+        "with_external_libs": [True, False]}
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_alsa": True,
+        "with_sqlite": True,
+        "with_external_libs": True,
+        "libalsa:shared": True}
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        if self.options.with_alsa:
+            self.requires("libalsa/1.1.9")
+        if self.options.with_sqlite:
+            self.requires("sqlite3/3.29.0")
+        if self.options.with_external_libs:
+            self.requires("flac/1.3.3")
+            self.requires("ogg/1.3.4")
+            self.requires("vorbis/1.3.6")
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("The project libsndfile can not be built by Visual Studio.")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.stdcpp
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            self._autotools = AutoToolsBuildEnvironment(self)
+            args = ["--without-caps"]
+            if not self.options.with_alsa:
+                args.append("--disable-alsa")
+            if not self.options.with_sqlite:
+                args.append("--disable-sqlite")
+            if not self.options.with_external_libs:
+                args.append("--disable-external-libs")
```

Comment:
I believe it's better to be explicit then to leave it up to the configure script.
```suggestion
            args = [
                "--without-caps",
                "--enable-alsa" if self.options.with_alsa else "--disable-alsa",
                "--enable-sqlite" if self.options.with_sqlite else "--disable-sqlite",
                "--enable-external-libs" if self.options.with_external_libs else "--disable-external-libs",
            ]
```